### PR TITLE
Remote Worker: Add tests for process_submission_message

### PR DESCRIPTION
**Pull request by nikochiko**
Add tests for remote submission worker

Additions proposed:
1. Unit test for `process_submission_message` function when the function is successful
2. Unit test for `process_submission_message` function when submission does not exist
3. Unit test for `process_submission_message` function when challenge phase does not exist

### CODE REVIEW

File: tests/unit/remoteworker/test_remote_worker.py [hunk 1]
```diff
@@ -212,3 +213,98 @@ def test_download_and_extract_file_when_download_fails(self, mock_logger):
...
+        self.input_file = "user_annotation_file.txt"
+        self.remote_evaluation = "Test Remote Evaluation"
+
+        # Creating sample outputs with minimal data for mocked functions
+        self.message = {
+            "challenge_pk": self.challenge_pk,
+            "phase_pk": self.challenge_phase_pk,
+            "submission_pk": self.submission_pk,
+        }
+        self.submission = {"id": self.submission_pk, "input_file": self.input_file}
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: @nikochiko can you add every new key value pair on a newline. Just to make the code clean.

File: tests/unit/remoteworker/test_remote_worker.py [hunk 2]
```diff
@@ -212,3 +213,98 @@ def test_download_and_extract_file_when_download_fails(self, mock_logger):
...
+        self.challenge = {"id": self.challenge_pk, "remote_evaluation": self.remote_evaluation}
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: same here

File: tests/unit/remoteworker/test_remote_worker.py (hunk 1 above)
Comment (nikochiko) [reply tests/unit/remoteworker/test_remote_worker.py]: Sure! Will do it right away!

**Comment by nikochiko** [comment]
@Ram81 I have made the changes.

**Pull request closed without merging**